src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/helper/DocumentHelper.java

```diff
@@ -345,7 +364,8 @@ public BulkPrintDocument getCaseDocumentAsBulkPrintDocument(CaseDocument caseDoc
 
         for (Map document : documentList) {
             Map value = (Map) document.get(VALUE);
-            Map<String, Object> documentLink = (Map) value.get(documentName);
+            Map<String, Object> documentLink = documentName != null ? (Map) value.get(documentName) : value;
```

Comment:
because you're passing null into this method surely this check will never return true and we'll never have a document link? might need a re-think of this implementation

Reply:
This method can also directly be accessed, and is used by the ConsentOrderApprovedDocumentService. I have created an `accessor` for the scenario where we are searching a collection that is a basic CCD configuration and noyt a custom complex type within a collection

Reply:
but if you're always passing null to this method from the accessor then it's never going to return any links so why are we calling it?

Reply:
I have given it an accessor for the scenario where we need to look up a basic CCD Collection (getCollectionOfDocumentLinksAsBulkPrintDocuments)
However the getDocumentLinksFromCustomCollectionAsBulkPrintDocuments method is still public, and is used by the ConsentOrderApprovedDocumentService, where it passes in the string of the document to be retrieved in the custom collection. Its probably easier to demo or show this to you on a call!

Reply:
yep, give me a shout

Reply:
I see now that we're setting the link to value rather than getting the document name, marking this as resolved